app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -267,6 +267,22 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
 
             Mage::helper('boltpay')->collectTotals($immutableQuote, true)->save();
 
+            ////////////////////////////////////////////////////////////////////////////
+            // reset increment id if needed
+            ////////////////////////////////////////////////////////////////////////////
+            /* @var Mage_Sales_Model_Order $preExistingOrder */
+            $preExistingOrder = Mage::getModel('sales/order')->loadByIncrementId($parentQuote->getReservedOrderId());
+
+            if (!$preExistingOrder->isObjectNew()) {
+                $parentQuote
+                    ->setReservedOrderId(null)
+                    ->reserveOrderId()
+                    ->save();
+
+                $immutableQuote->setReservedOrderId($parentQuote->getReservedOrderId());
+            }
+            ////////////////////////////////////////////////////////////////////////////
+
```

Comment:
I may be wrong, but it seems to me that the following two lines do the same as this block.

$parentQuote->reserveOrderId()->save(); // reset increment id is implemented as a part of reserveOrderId()
$immutableQuote->setReservedOrderId($parentQuote->getReservedOrderId());

Reply:
@danac-gs I would say that you are right, in theory.  That is certainly how I first implemented the code after looking at the above implementation of the `reserveOrderId` method.

In practice, however, at least on Swissgear, calling this method continued to hang onto reserved order id, most likely because they have over-ridden this for their special increment IDs that all start with SWG.

Swissgear aside, because we have already incurred the expense of doing the lookup for the order id and already know that it needs to be updated, you can think of the `->setReservedOrderId(null)` as a `-force` flag saying that we are not interested in checking if the currently assigned reserved order ID exists, we trust our logic that makes our plugin work and want a new one no matter what.